src/org/sosy_lab/java_smt/solvers/z3/Z3FormulaCreator.java

```diff
@@ -193,12 +196,16 @@ public Long extractInfo(Formula pT) {
       case Z3_RE_SORT:
       case Z3_UNKNOWN_SORT:
       case Z3_UNINTERPRETED_SORT:
-        // TODO: support for remaining sorts.
-        throw new IllegalArgumentException(
-            "Unknown formula type "
-                + Native.sortToString(z3context, pSort)
-                + " with sort "
-                + sortKind);
+        if (Native.isStringSort(z3context, pSort)) {
```

Comment:
Is there an enum value for `Z3_STRING_SORT`? I could only find `Z3_RE_SORT`, is this the same sort and would match here, too?